docs/specs/xref.yml

```diff
@@ -42,3 +42,65 @@ outputs:
     { "content": "<p>Link to <a href=\"https://docs.microsoft.com/en-us/dotnet/api/system.string\">String</a></p>\n"}
   build.manifest:
 ---
+# Define uid in conceptual, resolve it
+inputs:
+  docfx.yml: |
+    buildInternalXrefMap: true
+  docs/a.md: |
+    ---
+    title: Title from yaml header a
+    uid: a
+    ---
+  docs/b.md: |
+    ---
+    title: Title from yaml header b
+    uid: b
+    ---
+  docs/c.md: Link to @a
+outputs:
+  docs/c.json: | 
+    {"content":"<p>Link to <a href=\"docs/a.json\">Title from yaml header a</a></p>\n"}
+  docs/a.json:
+  docs/b.json:
+  build.manifest:
+---
+# Define uid in SDP, resolve it
+inputs:
+  docfx.yml: |
+    buildInternalXrefMap: true
+  docs/a.json: |
+    {
+      "$schema": "https://raw.githubusercontent.com/dotnet/docfx/v3/schemas/TestData.json",
+      "uid": "a",
+      "name": "test",
+      "fullName": "test.test",
+      "description": "a.description"
+    }
+  docs/c.md: Link to @a
+outputs:
+  docs/c.json: | 
+    {"content":"<p>Link to <a href=\"docs/a.json\">a</a></p>\n"}
+  docs/a.json:
+  build.manifest:
+---
+# Define duplicate uid in conceptual and SDP, should output warning
+inputs:
+  docfx.yml: |
+    buildInternalXrefMap: true
+  docs/a.yml: |
+    #YamlMime:TestData
+    uid: a
+  docs/b.json: |
+    {
+      "$schema": "https://raw.githubusercontent.com/dotnet/docfx/v3/schemas/TestData.json",
+      "uid": "a"
+    }
+  docs/c.md: Link to @a
+outputs:
+  docs/a.json:
+  docs/b.json:
+  docs/c.json:
+  build.manifest:
+  build.log: |
+    ["error","duplicate-uid","Uid 'a' has already been defined in *.","*"]
```

Comment:
can we make this error deterministic? similar to how `publish-url-conflict` works